src/jbi/configuration.py

```diff
@@ -0,0 +1,97 @@
+import json
+from inspect import signature
+from pathlib import Path
+from types import ModuleType
+from typing import Dict, Optional
+
+from src.app import environment
+from src.jbi import actions
+
+settings = environment.get_settings()
+
+
+class ConfigError(Exception):
+    pass
+
+
+class ProcessError(Exception):
+    pass
+
+
+def process_config_per_file(filename: str):
+    """
+        Process and validate (or error) for a single JBI config file
+    :param filename:
+    :return: dict of whiteboard_tag:json_file_values
+    """
+    default_dict = {
+        "action": "src.jbi.whiteboard_actions.default",
+        "enabled": True,
+    }  # Could be migrated to a json file
+    with open(filename, encoding="utf-8") as file:
+        data = json.load(file)
+        file_dict = default_dict.copy()
+        file_dict.update(data)
+
+        # Only process enabled files
+        if not file_dict.get("enabled"):
+            return None, None
+
+        # Confirm action exists
+        action = file_dict.get("action")
+        if action not in actions.module_dict.keys():
+            raise ConfigError(f"CONFIG ERROR: Unknown action `{action}`.")
```

Comment:
You could do this instead:

```py
try:
    action_module = importlib.import_module(action)
except ImportError as e:
    raise ConfigError(f"CONFIG ERROR: Unknown action `{action}`.")
``` 

And this way you can get rid of `module_dict`

Reply:
I think we can get rid of the whole `action.py` module this way too